hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -234,6 +237,18 @@ public void wipe(OwnershipTracker ownershipTracker, TokenRelationship accountRel
 		setTotalSupply(newTotalSupply);
 	}
 
+    public void delete() {
+        validateTrue(hasAdminKey(), TOKEN_IS_IMMUTABLE);
+        validateFalse(isDeleted(), TOKEN_WAS_DELETED);
+        validateFalse(isBelievedToHaveBeenAutoRemoved(), TOKEN_WAS_DELETED);
```

Comment:
I am not sure why we have this check? Can you point out where we execute it in the legacy code?